Sources/RxMoya/RxMoyaProvider.swift

```diff
@@ -21,7 +21,10 @@ open class RxMoyaProvider<Target>: MoyaProvider<Target> where Target: TargetType
 
         // Creates an observable that starts a request each time it's subscribed to.
         return Observable.create { [weak self] observer in
```

Comment:
What about killing the `[weak self]` here and just strongly capturing. Then we can kill the `guard`. There doesn't appear to be a risk of a retain cycle here, so the only other reason to do it this way would be if we *want* to use the release of an `RxMoyaProvider` to error out observables it has created. I think we want those `Observable`s to be valid until they are disposed, right?